scripts/codegen/CAN/App_CanMsgs.dbc

```diff
@@ -63,6 +65,36 @@ SG_ AIR_NEGATIVE : 1|1@1+ (1,0) [0|1] "" FSM,PDM
 BO_ 113 BMS_MOTOR_SHUTDOWN_ERRORS: 8 BMS
 SG_ DUMMY_MOTOR_SHUTDOWN : 0|1@1+ (1,0) [0|1] "" DEBUG
 
+BO_ 114 BMS_CELL_MONITOR_MIN_AND_MAX_VOLTAGES: 8 BMS
+SG_ MIN_CELL_VOLTAGE : 0|32@1+ (1,0) [3.0|4.20] "" DEBUG
+SG_ MAX_CELL_VOLTAGE : 32|32@1+ (1,0) [3.0|4.20] "" DEBUG
+
+BO_ 115 BMS_CELL_MONITOR_PACK_AND_AVERAGE_CELL: 8 BMS
+SG_ PACK_VOLTAGE : 0|32@1+ (1,0) [288.0|403.2] "" DEBUG
+SG_ AVERAGE_CELL_VOLTAGE : 32|32@1+ (1,0) [3.0|4.20] "" DEBUG
+
+BO_ 116 BMS_SEGMENT_0_AND_1: 8 BMS
+SG_ SEGMENT_0_VOLTAGE : 0|32@1+ (1,0) [48.0|67.2] "" DEBUG
+SG_ SEGMENT_1_VOLTAGE : 32|32@1+ (1,0) [48.0|67.2] "" DEBUG
+
+BO_ 117 BMS_SEGMENT_2_AND_3: 8 BMS
+SG_ SEGMENT_2_VOLTAGE : 0|32@1+ (1,0) [48.0|67.2] "" DEBUG
+SG_ SEGMENT_3_VOLTAGE : 32|32@1+ (1,0) [48.0|67.2] "" DEBUG
+
+BO_ 118 BMS_SEGMENT_4_AND_5: 8 BMS
+SG_ SEGMENT_4_VOLTAGE : 0|32@1+ (1,0) [48.0|67.2] "" DEBUG
+SG_ SEGMENT_5_VOLTAGE : 32|32@1+ (1,0) [48.0|67.2] "" DEBUG
+
+BO_ 119 BMS_CELL_VOLTAGE_ERRORS: 8 BMS
```

Comment:
I wanted to add these signals since I thought they would be helpful for debugging. Not sure what to name these errors though.

Reply:
Since the min and max cell voltages is already being used to set AIR_SHUTDOWN errors. I guess these could be labeled as non critical errors? I think it may slightly misleading since having a segment, pack or average cell that is under/overvoltage is a critical error. Open to ideas here.

Reply:
@kirons96

Reply:
under/over voltage has to be critical faults, even if we misread it, because of the fire danger

how would you distinguish between misreading and an  actual over/undervoltage event?

maybe by saving the past cell voltages at 1Hz and if you see over/undervoltage, make sure it's not by an excessive amount (eg. 1V difference)?

Reply:
Oh good point. Right now if the voltage is detected to be under 3.0V or above 4.2V then we transition into the fault state but it currently depends on the value of the min and max cell voltage read at that time. This would happen every second though. 

Since we are measuring values every 10ms I'm thinking we can use a moving average here for the max/min cell voltage values?

Reply:
Original question was more about asking what to name these signals though (CAN signals that aren't min/max cell voltages)

Reply:
there's already an RC filter in hardware, so we should be careful about not filtering too aggressively in firmware

if we have a firmware filter, then we probably don't need cell voltage error detection + this CAN message

Reply:
i mean it's a legit solution to the problem of a bad reading, but those are highly improbable. the hardware filter won't fix a single/double bit error right, assuming it either gets past the CRC (lol) or the error occurs at the MCU (?)

not sure if this problem is even worth considering - take some time to think about how you'd actually get a bad reading and what the probabilities are